src/neuroglancer/user_layer_with_mip_level_constraints.ts

```diff
@@ -0,0 +1,81 @@
+import {UserLayer} from 'neuroglancer/layer';
+import {TrackableMIPLevelConstraints} from 'neuroglancer/trackable_mip_level_constraints';
+import {UserLayerWithVolumeSource, UserLayerWithVolumeSourceMixin} from 'neuroglancer/user_layer_with_volume_source';
+import {RenderLayer as GenericSliceViewRenderLayer} from 'neuroglancer/sliceview/renderlayer.ts';
+import {vec3} from 'neuroglancer/util/geom';
+import { Tab } from 'neuroglancer/widget/tab_view';
+import { VoxelSizeSelectionWidget } from 'neuroglancer/widget/voxel_size_selection_widget';
+
+const MIN_MIP_LEVEL_JSON_KEY = 'minMIPLevel';
+const MAX_MIP_LEVEL_JSON_KEY = 'maxMIPLevel';
+
+function helper<TBase extends {new (...args: any[]): UserLayerWithVolumeSource}>(Base: TBase) {
+  class C extends Base implements UserLayerWithMIPLevelConstraints {
```

Comment:
The MIP restrictions are very closely related to `UserLayerWithVolumeSource`. I think instead of creating a new Mixin, it is reasonable to just add the new functionality to the existing `UserLayerWithVolumeSource` in this case.